subsys/bluetooth/audio/unicast_client.c

```diff
@@ -730,12 +730,15 @@ static void unicast_client_ep_releasing_state(struct bt_audio_ep *ep,
 
 	LOG_DBG("dir %s", bt_audio_dir_str(ep->dir));
 
-	/* The Unicast Client shall terminate any CIS established for that ASE
-	 * by following the Connected Isochronous Stream Terminate procedure
-	 * defined in Volume 3, Part C, Section 9.3.15 in when the Unicast
-	 * Client has determined that the ASE is in the Releasing state.
-	 */
-	bt_audio_stream_disconnect(stream);
+	if (ep->iso != NULL && ep->iso->chan.state == BT_ISO_STATE_CONNECTED) {
```

Comment:
Shouldn't disconnect be called as well if `BT_ISO_STATE_CONNECTING` like you did in other places? Going from Enabling to Releasing I guess there might be no CIS Connected yet (looking at Figure 3.2 in ASCS)

Reply:
The addition check for `BT_ISO_STATE_CONNECTING` is added an a later commit in this PR :) At the time of this commit, no other places checks it.